umbraco/Umbraco-CMS
# dotnet new template - Escape JSON special characters for password & connection string

This PR is to fix #10424 where if the dotnet new CLI options for connectionstring & password contained `\` or `"` in a JSON string inside the appsettings.development.json file would break the JSON file.

This is a bug to help fix using the dotnet new template with unattended install to help skip the steps in the UI installer and go straight to installing and configuring Umbraco on its first run.

## Solution
To use Regex to replace the user input value from the connection string and password parameters using the Regex Symbol Generator
https://github.com/dotnet/templating/wiki/Available-Symbols-Generators#regex

## Testing
* Run build.ps1 in build folder
* Uninstall previous template pack `dotnet new -u Umbraco.Templates`
* Change Dir to `build.out` install new template pack `dotnet new -i .\Umbraco.Templates.9.0.0-beta003.nupkg`
* Run CLI command below
* Verify output in `Appsettings.Development.json`
* Create new project with Visual Studio UI

## Example CLI command
`dotnet new umbraco --friendly-name 'Warren Buckley' --email warren@umbraco.com --connection-string 'Server=(localdb)\mssqllocaldb;Database=U9beta3;Integrated Security=true' --password 'my\"password\will-break-json' --name testy-test-project`

## Example Values
**Connection String**
`Server=(localdb)\mssqllocaldb;Database=U9beta3;Integrated Security=true`

**Password**
`my"password\will-break-json`

![image](https://user-images.githubusercontent.com/1389894/121904074-caccd080-cd20-11eb-9812-2bc6be26608b.png)
